### PR TITLE
Always enable clustering

### PR DESCRIPTION
**Pull request by yim-lee**
Resolves https://github.com/apple/swift-distributed-actors/issues/880

### CODE REVIEW

**Comment by yim-lee** [comment]
Seeing `Fatal error: Somehow attempted to load system.cluster before it was initialized!` in different tests and not sure how to track down the cause.

**Comment by ktoso** [comment]
> Seeing Fatal error: Somehow attempted to load system.cluster before it was initialized! in different tests and not sure how to track down the cause.

I've seen that a few times and track it as https://github.com/apple/swift-distributed-actors/issues/840

It is about the actor starting "early" before the cluster var is initialized; this should not ever happen, so I'll look into it. If you see that failure ignore it for now

File: Tests/DistributedActorsTests/Cluster/MembershipGossipLogicSimulationTests.swift [hunk 1]
```diff
@@ -19,10 +19,6 @@ import NIO
 import XCTest
 
 final class MembershipGossipLogicSimulationTests: ClusteredActorSystemsXCTestCase {
-    override func configureActorSystem(settings: inout ClusterSystemSettings) {
-        settings.enabled = false // not actually clustering, just need a few nodes
-    }
```
Comment (ktoso) [inline Tests/DistributedActorsTests/Cluster/MembershipGossipLogicSimulationTests.swift:24]: Eh Oh Interesting... I guess maybe we do want to allow for this just of our internal tests then... :-\ 

Will think some more.

**Comment by ktoso** [comment]
Actually let's merge -- and I'll fix one test after this

File: Tests/DistributedActorsTests/TraversalTests.swift [hunk 2]
```diff
@@ -82,7 +82,9 @@ final class TraversalTests: ActorSystemXCTestCase {
                 "system",
                 "receptionist",
                 "receptionist-ref",
-                // "gossip", // TODO: should it also be here?
+                // cluster actors ---
+                "downingStrategy", "swim", "nodeDeathWatcher", "gossip", "cluster", "leadership",
+                // end of cluster actors ---
```
Comment (ktoso) [inline Tests/DistributedActorsTests/TraversalTests.swift:87]: this test needs to account for the cluster actors now 👍 

Seems all others do pass, at least locally for me

**Review by ktoso — APPROVED** [review]
This is LGTM, I'll merge this in a moment, just need to phase merges

File: Tests/DistributedActorsTests/TraversalTests.swift (hunk 2 above)
Comment (yim-lee) [reply Tests/DistributedActorsTests/TraversalTests.swift:87]: 🙏

**Pull request merged**